src/rantanplan/structures.py

```diff
@@ -258,314 +178,320 @@ def is_terceto_encadenado(rhyme_pattern):
-        is_seguidilla
+        lambda ranges_list: has_fixed_length_verses(STRUCTURES_LENGTH["seguidilla"], ranges_list, allow_fluctuation=True)
     ), (
         CONSONANT_RHYME,
         "seguidilla_compuesta",
         r"((-a-a)|(abab))((a-a)|(b-b)|(c-c))",
-        is_seguidilla_compuesta
+        lambda ranges_list: has_fixed_length_verses(STRUCTURES_LENGTH["seguidilla_compuesta"], ranges_list)
     ), (
         ASSONANT_RHYME,
         "seguidilla_compuesta",
         r"((-a-a)|(abab))((a-a)|(b-b)|(c-c))",
-        is_seguidilla_compuesta
+        lambda ranges_list: has_fixed_length_verses(STRUCTURES_LENGTH["seguidilla_compuesta"], ranges_list)
     ), (
         ASSONANT_RHYME,
         "chamberga",
         r"((-a-a)|(abab))([^-]{2}){3}",
-        is_chamberga
+        lambda ranges_list: has_fixed_length_verses(STRUCTURES_LENGTH["chamberga"], ranges_list)
     ), (
         ASSONANT_RHYME,
         "seguidilla_gitana",
         r"(-a-a)|(a-a-)",
-        is_seguidilla_gitana
+        lambda ranges_list: has_fixed_length_verses(STRUCTURES_LENGTH["seguidilla_gitana"], ranges_list)
     ), (
         ASSONANT_RHYME,
         "endecha_real",
         r"(-a-a){2,}",
-        is_endecha_real
+        lambda ranges_list: has_fixed_length_verses(STRUCTURES_LENGTH["endecha_real"], ranges_list, repeating_structure=True)
     ), (
         CONSONANT_RHYME,
         "cuarteto_lira",
         r"(abab)|(abba)|(-a-a)",
-        lambda lengths: all(
-            x in most_common_lengths(lengths, n=3, number_of_verses=4) for x in
-            [11, 7])
+        lambda ranges_list: has_mixed_length_verses(LIRA_LONG_LINE, LIRA_SHORT_LINE, ranges_list)
     ), (
         ASSONANT_RHYME,
         "cuarteto_lira",
         r"(abab)|(abba)|(-a-a)",
-        lambda lengths: all(
-            x in most_common_lengths(lengths, n=3, number_of_verses=4) for x in
-            [11, 7])
+        lambda ranges_list: has_mixed_length_verses(LIRA_LONG_LINE, LIRA_SHORT_LINE, ranges_list)
     ), (
         CONSONANT_RHYME,
         "estrofa_sáfica",  # se puede encadenar?
         r"(----)|(a-a-)|(ab-b)|(abab)",
-        is_estrofa_safica or is_estrofa_safica_unamuno
+        lambda ranges_list: (
+            has_fixed_length_verses(
+                STRUCTURES_LENGTH["estrofa_safica"], ranges_list)
+                        ) or (
+            has_fixed_length_verses(
+                STRUCTURES_LENGTH["estrofa_unamuno"], ranges_list))
     ), (
         ASSONANT_RHYME,
         "estrofa_sáfica",  # se puede encadenar?
         r"(----)|(a-a-)|(ab-b)|(abab)",
-        is_estrofa_safica
+        lambda ranges_list: has_fixed_length_verses(STRUCTURES_LENGTH["estrofa_safica"], ranges_list)
     ), (
         CONSONANT_RHYME,
         "estrofa_francisco_de_la_torre",
         r"(----)|(a-a-)",
-        is_estrofa_francisco_de_la_torre
+        lambda ranges_list: has_fixed_length_verses(STRUCTURES_LENGTH["estrofa_francisco_de_la_torre"], ranges_list)
```

Comment:
One option to make these lists shorter is to pass in just the key to `STRUCTURES_LENGTH` and have the function recover the corresponding value.

Reply:
Will look onto that

Reply:
Done :)